mongospecs/mixins/crud.py

```
@@ -0,0 +1,323 @@
+    def find_one(cls, filter: FilterType = None, **kwargs: t.Any) -> SpecDocumentType:
+        """Return the first document matching the filter"""
+        # Flatten the projection
+        kwargs["projection"], references, subs = cls._flatten_projection(
+            kwargs.get("projection", cls._default_projection)
+        )
+
+        # Find the document
+        if isinstance(filter, (Condition, Group)):
+            filter = filter.to_dict()
+
+        coll: Collection[SpecDocumentType] = cls.get_collection()
```

Ruff (0.8.2)

Undefined name `Collection`

(F821)